ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/DdlCommandExec.java

```diff
@@ -66,6 +66,16 @@ private Executor(final String sql, final boolean withQuery) {
 
     @Override
     public DdlCommandResult executeCreateStream(final CreateStreamCommand createStream) {
+      final SourceName sourceName = createStream.getSourceName();
+      final DataSource dataSource = metaStore.getSource(sourceName);
+
+      if (dataSource != null && !createStream.isOrReplace()) {
+        return new DdlCommandResult(true,
+            String.format("Cannot add stream %s: A stream with the same name "
+                    + "already exists.",
+                sourceName));
+      }
```

Comment:
Isn't this check now redundant, given there is code in `CreateSourceFactory` to do this now?  We should let the code do what it used to do, which is throw an exception, not return a success code!

Reply:
The code in `CreateSourceFactory` checks if a `dataSource`  already exists , if `isOrReplace()` is not set and `ifNotExists` is also not set and throws exception , which seems right
If the code passes the checks in that gate ,it means in `DdlCommandExec` it would mean the ifNotExists flag is set to true and we need to ensure that the statement does NOT fail ?

Reply:
Seems redundant, but it is what DROP IF NOT EXIST does too. It has a check in the DropSourceCommand and another in this class, which returns the DdlCommandResult message without causing a failure.

Reply:
I figured out why we need this repeated code. Turns out that `CreateSourceFactory` is only called for `CREATE` without queries statements. For `CREATE AS`, a different code path is called, which creates the Create command similar to the `CreateSourceFactory`, but without the factory validations and with more parameters. That path is `EngineExecutor.maybeCreateSinkDdl`. 

I was thinking on refactor the code, but seems it requires several changes. For now, I leave this temporary and do a check inside the `maybeCreateSinkDdl` for the `isNotExists` value. I'll do the refactor in another PR.